### ISSUE TITLE
Missing MapboxMobileEvents.framework/Info.plist

### ISSUE THREAD

**Issue by acoulon99** [rnmapbox/maps#1649]
**Describe the bug**
After following the @react-native-mapbox-gl/maps installation instructions for iOS, the app was unable to be installed on a simulator because of a missing Info.plist file for MapboxMobileEvents.

**To Reproduce** 
Steps to reproduce the behavior. 

1. Create project and install @react-native-mapbox-gl/maps

```sh
react-native init MapboxMobileEventsBugReproduction --version 0.66.3
cd MapboxMobileEventsBugReproduction
npm install @react-native-mapbox-gl/maps@8.5.0 --save
```

2. Modify Podfile by ensuring the following hooks exist
```
  pre_install do |installer|
    $RNMBGL.pre_install(installer)
    [ ... ]
  end

  post_install do |installer|
    $RNMBGL.post_install(installer)
    [ ... ]
  end
```

3. Install Pods
```sh
cd ios && pod install && cd ..
```

4. Open Xcode, target an iPhone 11 simulated device and click the "Start the active scheme" play button (Xcode)

**Expected behavior** 
MapboxMobileEventsBugReproduction installs and launches successfully on the simulated device.

**Actual behavior** 
After a successful build, while installing the app onto an iOS simulated device, a popup within Xcode is displayed saying that  the app cannot be installed because of a missing Info.plist file within MapboxMobileEvents.

**Screenshots** 

Error popup
<img width="263" alt="Screen Shot 2021-11-28 at 9 31 39 PM" src="https://user-images.githubusercontent.com/9921114/143805114-073be9b6-4d41-4caa-b9ed-18ce20407b71.png">

No .plist file within "MapboxMobileEvents/Support Files"
<img width="296" alt="Screen Shot 2021-11-28 at 9 35 36 PM" src="https://user-images.githubusercontent.com/9921114/143805128-65bc0297-7bf8-4098-9213-768b7f4debed.png">

**Versions** 
 - Platform: iOS
 - Platform OS: iOS 15
 - Device: iPhone 12
 - Emulator/ Simulator: yes
 - Dev OS: macOS 11.6 BigSur
 - react-native-mapbox-gl Version: 8.5.0
 - Mapbox GL version: Mapbox-iOS-SDK 5.9.0
 - React Native Version: 0.66.3
 - Cocoapods: 1.11.2

**Additional context**
<details>
<summary>Simulator installation error</summary>

```
Details

Unable To Install “MapboxMobileEventsBugReproduction”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework; Extra info about "/Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework/Info.plist": Couldn't stat /Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework/Info.plist: No such file or directory
User Info: {
    IDERunOperationFailingWorker = IDELaunchiPhoneSimulatorLauncher;
}
--
Failed to load Info.plist from bundle at path /Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework; Extra info about "/Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework/Info.plist": Couldn't stat /Users/puma/Library/Developer/CoreSimulator/Devices/22B466BB-CDA3-4124-BEDA-11F4F2E892AE/data/Library/Caches/com.apple.mobile.installd.staging/temp.qGbyjM/extracted/Payload/MapboxMobileEventsBugReproduction.app/Frameworks/MapboxMobileEvents.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 47;
}
--

Analytics Event: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_model" = "iPhone13,2";
    "device_osBuild" = "15.0 (19A339)";
    "device_platform" = "com.apple.platform.iphonesimulator";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = "x86_64";
    "operation_duration_ms" = 2012;
    "operation_errorCode" = 1;
    "operation_errorDomain" = IXUserPresentableErrorDomain;
    "operation_errorWorker" = IDELaunchiPhoneSimulatorLauncher;
    "operation_name" = IDERunOperationWorkerGroup;
    "param_consoleMode" = 0;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 3;
    "param_destination_isProxy" = 0;
    "param_destination_platform" = "com.apple.platform.iphonesimulator";
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 0;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 0;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 0;
    "param_launcher_substyle" = 0;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_runnable_swiftVersion" = "5.5.1";
    "param_runnable_type" = 2;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphonesimulator15.0";
    "sdk_osVersion" = "15.0";
    "sdk_variant" = iphonesimulator;
}
--


System Information

macOS Version 11.6 (Build 20G165)
Xcode 13.1 (19466) (Build 13A1030d)
Timestamp: 2021-11-28T21:32:11-06:00
```
</details>

<details>
<summary>Podfile</summary>

```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '11.0'

target 'MapboxMobileEventsBugReproduction' do
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => false
  )

  target 'MapboxMobileEventsBugReproductionTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  use_flipper!()

  pre_install do |installer|
    $RNMBGL.pre_install(installer)
  end

  post_install do |installer|
    $RNMBGL.post_install(installer)
    react_native_post_install(installer)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end
```
</details>

**Replicates**
https://github.com/react-native-mapbox-gl/maps/issues/1415

**Seemingly related**
https://github.com/mapbox/mapbox-gl-native-ios/issues/555

**Tried**
- Clearing Cococapod cache
- Reintegrating Cocoapods
- Reinstalling Cocoapods
- Clearing ios build directory
- Clearing DerivedData directory
- `/example` project (same result)

**Comment by tomgreco** [comment]
This issue has been haunting me for weeks. Usually clearing Pods, Podfile.lock, and doing a rain dance fixes it. But lately no luck at all.

**Comment by acoulon99** [comment]
@tomgreco so... perhaps a recent rain dance fixed the issue because it's working for me now 😂

After reading this comment and inspecting the Pods, it seems as if the root cause could be related to "MapboxMobileEvents-Info.plist" missing from "ios/Pods/Target Support Files/MapboxMobileEvents" and the XCode project, which could potentially happen again.

https://github.com/mapbox/mapbox-gl-native-ios/issues/555#issuecomment-986061599

**Comment by ferdicus** [comment]
hey @tomgreco, I'm no longer actively maintaining this repo, see [my post here](https://github.com/react-native-mapbox-gl/maps/discussions/1680#discussion-3770355)

**Comment by haron68** [comment]
> > @tomgreco so... perhaps a recent rain dance fixed the issue because it's working for me now 😂
> > After reading this comment and inspecting the Pods, it seems as if the root cause could be related to MapboxMobileEvents-Info.plist missing from "ios/Pods/Target Support Files/MapboxMobileEvents" and the XCode project, which could potentially happen again.
> > https://github.com/mapbox/mapbox-events-ios/blob/main/Sources/MapboxMobileEvents/Info.plist
> > [mapbox/mapbox-gl-native-ios#555 (comment)](https://github.com/mapbox/mapbox-gl-native-ios/issues/555#issuecomment-986061599)
> 
> @acoulon99 I am jealous. The issue has now come back to haunt me again! Tried my usual fix but doesn't seem to be working anymore.
> 
> ```
> rm -rf ios/Pods ios/build ~/Library/Caches/CocoaPods ~/Library/Developer/Xcode/DerivedData && pod cache clean --all
> npm i --legacy-peer-deps && cd ios && pod install
> ```
> 
> @ferdicus here is how to recreate the issue: [mapbox/mapbox-gl-native-ios#555 (comment)](https://github.com/mapbox/mapbox-gl-native-ios/issues/555#issuecomment-986061599)
> 
> Can you see if anything can be done about this?

I also just started seeing this problem have you tried downgrading versions from `8.5.0` -> `8.3.0`?

**Comment by tomgreco** [comment]
> I also just started seeing this problem have you tried downgrading versions from `8.5.0` -> `8.3.0`?

@haron68 for now I have a copy of the Pods folder from a working build saved on my desktop to revert back to whenever this happens.

**Comment by haron68** [comment]
@tomgreco the way I fixed mine is by manually adding [this file](https://github.com/mapbox/mapbox-events-ios/blob/main/Sources/MapboxMobileEvents/Info.plist) to the `Support Files` in the MapboxMobileEvents pod and then in the target I have to add the file by pressing the `"Choose Info.plist file..."` button then it shows this after and builds again

<img width="786" alt="image" src="https://user-images.githubusercontent.com/27697689/153689012-af32427c-c2f3-4d5c-959a-a2fb8d37736c.png">

**Comment by djpane** [comment]
I bumped into this issue about six months ago and now again all of the sudden. It seems to be completely random since I'm using Mapbox in four different RN projects and fixing is seems to be random as well. Don't remember what magic I used the last time but this time after a day of banging my head to the wall I was able to run my project by doing a pod clean, pod deintegrate, delete podfile.lock, restarted my computer, cleaned the build folder and deleted derived data. I did all of those steps more than once before except restarting my machine in between. So who knows it required that as well. Or maybe it was the rain dance..
